CMakeLists.txt

```diff
@@ -1,16 +1,22 @@
-cmake_minimum_required(VERSION 2.8)
-project(azmq)
+cmake_minimum_required(VERSION 3.14)
+project(azmq VERSION 1.0.3 LANGUAGES CXX)
 
-include(AzmqCPack.cmake)
+list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake)
 
-set(CMAKE_CXX_STANDARD 11)
-set(CMAKE_CXX_STANDARD_REQUIRED ON)
-set(CMAKE_CXX_EXTENSIONS OFF)
-set(CMAKE_XCODE_ATTRIBUTE_CLANG_CXX_LANGUAGE_STANDARD "c++11")
+include(GNUInstallDirs)
+include(AzmqCPack)
 
-if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU" OR CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall")
-endif()
+find_package(Boost 1.48 COMPONENTS system date_time thread chrono random REQUIRED)
+find_package(ZeroMQ 4.0 REQUIRED)
```

Comment:
I would like to see a check for either libzmq or libzmq-static, which could be defined if both azmq and zeromq are being brought in as dependencies by the wrapping project using FetchContent.  cppzmq does something this, though I simplified it here.
```
if (NOT TARGET libzmq AND NOT TARGET libzmq-static)
    find_package(ZeroMQ 4.0 REQUIRED)
endif()
```

Reply:
Hi @rhvonlehe , thank you for your for the feedback. 

IMO it would be easier if libzmq provided just one target name for both static and shared versions, then the let it's users select what they want using the standard cmake flag  'BUILD_SHARED_LIBS`... but this isn't the case instead it output  2 target with different names, and this complication then flows onto consuming project like azmq.

In the MR, I've tried to set it up so you can, have the choice of  what to use

The logic is here
```
if (TARGET Azmq::libzmq)
    # allow overriding from the calling script
    message(STATUS "using provided Azmq::libzmq target")
elseif (TARGET libzmq AND BUILD_SHARED_LIBS)
    message(STATUS "using existing shared library target")
    add_library(Azmq::libzmq ALIAS libzmq)
elseif (TARGET libzmq-static AND NOT BUILD_SHARED_LIBS)
    message(STATUS "using existing static library target")
    add_library(Azmq::libzmq ALIAS libzmq-static)
else ()
    # try finding the package
    find_package(ZeroMQ QUIET)
```

1. You can define an alias target for `Azmq::libzmq` and this will be used first, this is the most flexible, you get to tell me what you want to use, and I'll use it.
2.  if you haven't told me, I then try with existing zmq targets if they exist
  2a. first to the shared lib
  2b. then to the static lib, and I honor the cmake flag BUILD_SHARED_LIBS
3. still no luck, I look for installed version of ZeroMq, and try again
  3a. first to the shared lib
  3b. then to the static lib
    
We could take the approach cppzmq follows and export both static and shared target for azmq, but I was attempted to make it easier for user by allowing them the choice with standard cmake option "BUILD_SHARED_LIBS" .

Reply:
This sounds perfectly reasonable to me - thanks for the update.  I agree about the use of two target names by zeromq.  It would be easier if it were one.

Reply:
@rhvonlehe , please review the updated cmake setup on master.